apis/core.oam.dev/common/types.go

```diff
@@ -476,6 +481,48 @@ const (
 	WorkflowResourceCreator ResourceCreatorRole = "workflow"
 )
 
+// OAMObjectReference defines the object reference for an oam resource
+type OAMObjectReference struct {
+	Component string `json:"component,omitempty"`
```

Comment:
Is it based on `one application--one component--one trait` module?

Reply:
Not exactly. This reference is for one target resource, for example, a deployment or a service. Each resource is always belonging to `1 app - 1 comp (- 1 trait, optional)`.